Fix AttributeError isAlive

BrutusTT opened the pull request:
starting roscore results in:

```
exception in shutdown_process_monitor: 'ProcessMonitor' object has no attribute 'isAlive'
Traceback (most recent call last):
  File "/Users/brutustt/ros_catkin_ws/install_isolated/lib/python3.9/site-packages/roslaunch/pmon.py", line 116, in shutdown_process_monitor
    if process_monitor.isAlive():
AttributeError: 'ProcessMonitor' object has no attribute 'isAlive'
```

However, the ProcessMonitor has a `is_alive` method so I assume this is what was meant here.

The same happens after closing rqt:

```
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/Users/ingo/ros_catkin_ws/src/ros_comm/rospy/src/rospy/core.py", line 615, in _ros_atexit
    signal_shutdown('atexit')
  File "/Users/ingo/ros_catkin_ws/src/ros_comm/rospy/src/rospy/core.py", line 598, in signal_shutdown
    if t.isAlive():
AttributeError: 'Thread' object has no attribute 'isAlive'
```

However, the Thread has a `is_alive` method so I assume this is what was meant here.

BrutusTT committed:
fix: roscore -> AttributeError: 'ProcessMonitor' object has no attribute 'isAlive'

BrutusTT committed:
Fix AttributeError: 'Thread' object has no attribute 'isAlive'

pschmutz commented:
There are still a few usages of `isAlive` in the code:
```
tools/roslaunch/test/unit/test_roslaunch_pmon.py:57:    def isAlive(self):
tools/roslaunch/test/unit/test_roslaunch_pmon.py:305:        self.assert_(pmon.isAlive())
tools/roslaunch/test/unit/test_roslaunch_pmon.py:307:        self.failIf(pmon.isAlive())
tools/roslaunch/test/unit/test_roslaunch_pmon.py:324:        self.assert_(pmon1.isAlive())
tools/roslaunch/test/unit/test_roslaunch_pmon.py:325:        self.assert_(pmon2.isAlive())        
tools/roslaunch/test/unit/test_roslaunch_pmon.py:329:        self.failIf(pmon1.isAlive())
tools/roslaunch/test/unit/test_roslaunch_pmon.py:330:        self.failIf(pmon2.isAlive())        
clients/rospy/src/rospy/core.py:512:            if not other.isAlive():
```
It probably makes sense to change them as well.

I am not sure if the `isAlive` function in the `ProcessMonitorMock` is used externally. So it might make sense to keep it and just add an `is_alive` or create an alias for it.

jacobperron commented:
Thanks for the fix!

I agree with @pschmutz about updating the remaining instances as well.

> I am not sure if the isAlive function in the ProcessMonitorMock is used externally. So it might make sense to keep it and just add an is_alive or create an alias for it.

+1 for keeping it and adding a new `is_alive` as needed.

@BrutusTT Do you mind updating this PR, replacing the other instances of isAlive as well?

jacobperron committed:
Update remaining instances of isAlive to is_alive

Keep mock method isAlive to keep backwards compatibility for users who happen to be using the mock object.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

jacobperron commented:
I went ahead an updated the remaining instances of 'isAlive' (0efe27b).
